wandb/sdk/lib/printer.py

```diff
@@ -439,25 +443,11 @@ def display(
         *,
         level: str | int | None = None,
     ) -> None:
-        text = "<br/>".join(text) if isinstance(text, (list, tuple)) else text
-        self._display_fn_mapping(level)(text)
-
-    @staticmethod
-    def _display_fn_mapping(level: str | int | None) -> Callable[[str], None]:
-        level = Printer._sanitize_level(level)
+        if wandb.run and wandb.run._settings.silent:
+            return
```

Comment:
hmmm... is it good idea to add the global print logic here? maybe just have some on/off condition outside of this?

Reply:
Note that `display_html` had this check before, so it's necessary here to preserve semantics. Either way, I think it makes sense to have it here: the caller decides what to print and when, and this module decides how and whether.

I don't like the `wandb.run` check---we should have package-level settings.

Reply:
yeah that's my issue with it the `wandb.run` usage, but if it was there already i'm ok keeping it for now, but i think having some TODO to remove it would be good

Reply:
can you move it up to the call-sites of this? instead of having it in the printer library?

Reply:
I think it needs to be in the printer library. The callsites shouldn't be able to ignore the `silent` setting.

Reply:
ok, if someone uses the printer library outside of the context of a run does it mean we still need to respect the silent of this run? i think we are mixing things here

Reply:
That's how it was before, and this PR should keep the behavior the same because more thought is needed to change it safely.

FWIW, I think it should be a package level setting, not a run setting, but we don't have that mechanism as far as I know.

Reply:
fine I’ll approve but I don’t agree with this  being here

Reply:
thanks! but just to clarify again: this was already here (display_fn_mapping called display_html, which contained this code)

Reply:
ok i went back in the code and found that change, but i still believe we need to remove this from this module in a future PR.